### PR TITLE
Features/7631

### PR DESCRIPTION
**Pull request by FuriousLlama**
[AB#7631](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/7631)

Removed user info from storage
Added validation to registration form
Wired up registration page to backend
Added authentication checks to webclient backend
Rewired vue front end to not use user info storage

### CODE REVIEW

**Comment by healthgatewaybc** [comment]
Common Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=commonlibrary_pr_213)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_213)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_213)

**Comment by healthgatewaybc** [comment]
Database Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=databaselibrary_pr_213)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_213)

File: Apps/WebClient/src/ClientApp/app/router.ts [hunk 1]
```diff
@@ -83,24 +84,13 @@ router.beforeEach(async (to, from, next) => {
       if (!hasAccess) {
         next({ path: "/login", query: { redirect: to.path } });
       } else {
-        let userIsRegistered: boolean = store.getters["auth/userIsRegistered"];
         // If the user is registerd and is attempting to go to registration re-route
+        let userIsRegistered: boolean = store.getters["user/userIsRegistered"];
         if (userIsRegistered && to.path === "/registration") {
```
Comment (TiagoGraf89) [inline Apps/WebClient/src/ClientApp/app/router.ts]: I think we need to include a check for the path /registrationInfo here too, its also part of the registration flow, if already registered go to timeline too

Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/app/router.ts]: added that path

**Comment by healthgatewaybc** [comment]
Database Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=databaselibrary_pr_213)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_213)

**Comment by healthgatewaybc** [comment]
WebClient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=webclient_pr_213)

**Pull request merged**